### PR TITLE
test(staging): add strip_template_suffix edge cases and fix misleading test name

### PR DESCRIPTION
**Pull request by scotthamilton77**
## Summary
- Adds three edge-case tests for `strip_template_suffix` (closes w1qls.3.4):
  - Embedded `.template` as a non-final suffix (e.g. `AGENTS.md.template.bak`) — path unchanged, only the final suffix is tested
  - Double `.template` suffix (e.g. `file.template.template`) — outermost only stripped, yielding `file.template`
  - No suffix at all (e.g. `Makefile`) — path unchanged
- Renames `test_build_plan_collects_all_phases` → `test_build_plan_includes_shared_and_tool_phases` (closes w1qls.3.5): the old name implied exhaustive phase enumeration; the test deliberately avoids a `len()` assertion to stay non-brittle as phases are added

## Test Plan
- [x] `make ci-installer` — 123 tests passing, 99.58% branch coverage, ruff ✅, mypy strict ✅, pip-audit clean ✅
- [x] New tests verified to exercise distinct coded decisions (not stdlib tautologies): embedded suffix and double-suffix tests guard against plausible wrong implementations; no-suffix test is borderline but documents intent for a real filename the installer encounters

## Scope
Test-only changes — no production code modified.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/installer/tests/unit/test_staging_build_plan.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def _make_repo(tmp_path: Path) -> Path:
     return repo
 
 
-def test_build_plan_collects_all_phases(tmp_path: Path) -> None:
+def test_build_plan_includes_shared_and_tool_phases(tmp_path: Path) -> None:
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply packages/installer/tests/unit/test_staging_build_plan.py:44]: Fixed in 44a3653ee57ebb116c0400a14fd0c40a6222eae4. Renamed stale test reference test_build_plan_collects_all_phases -> test_build_plan_includes_shared_and_tool_phases in staging plan doc (line 570)
